korangar/src/input/mod.rs

```diff
@@ -312,6 +316,18 @@ impl InputSystem {
             self.mouse_input_mode = MouseInputMode::RotateCamera;
         }
 
+        if self.right_mouse_button.pressed() && self.previous_mouse_button.0 == MouseButton::Right && client_tick.0.saturating_sub(self.previous_mouse_button.1.0) < DOUBLE_CLICK_TIME_MS {
+            events.push(UserEvent::CameraResetRotation);
+        }
+
+        if self.right_mouse_button.pressed() && !matches!(self.mouse_input_mode, MouseInputMode::RotateCamera) {
```

Comment:
I'm trying to prevent a right-click-drag (when we rotate the camera) to count as the first click.

Reply:
Testing it locally, I think it works fine. Do you think the current approach need further refinment?